galsim/chromatic.py

```diff
@@ -1677,12 +1758,15 @@ def __init__(self, *args, **kwargs):
         # For the case of ChromaticSums, as long as each object in the sum is separable (even if the
         # entire object is not) then interpolation can still be used.  So we do not warn about this
         # here.
-        n_nonsep = 0
-        n_interp = 0
-        for obj in self.objlist:
-            if not obj.separable and not isinstance(obj, galsim.ChromaticSum): n_nonsep += 1
-            if isinstance(obj, InterpolatedChromaticObject): n_interp += 1
-        if n_nonsep>1 and n_interp>0:
+        # n_nonsep = 0
+        # n_interp = 0
+        # for obj in self.objlist:
+        #     if not obj.separable and not isinstance(obj, galsim.ChromaticSum): n_nonsep += 1
+        #     if isinstance(obj, InterpolatedChromaticObject): n_interp += 1
+
+        n_nonsep = sum(o.separable and not isinstance(o, galsim.ChromaticSum) for o in self.objlist)
```

Comment:
This still has `o` as a variable.  IMO, `o` and `l` should never be used as variables.